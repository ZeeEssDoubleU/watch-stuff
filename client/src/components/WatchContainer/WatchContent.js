import React from "react";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import PropTypes from "prop-types";

import "./WatchContent.scss";
import Video from "../Video/Video";
import RelatedVideos from "../RelatedVideos/RelatedVideos";
import VideoInfo from "../Video/VideoInfo";
import VideoMetadata from "../Video/VIdeoMetadata";
import CommentsContainer from "../CommentsContainer/CommentsContainer";

import * as watchActions from "../../store/actions/watch";
import {
	selector_videoById,
	selector_relatedVideos,
} from "../../store/reducers/videos";
import { selector_channelDetails } from "../../store/reducers/channels";

const WatchContent = props => (
	<div className="watch-grid">
		<Video id={props.match.params.videoId} className="video-container" />
		<VideoMetadata className="video-metadata" video={props.video} />
		<VideoInfo
			className="video-info-container"
			video={props.video}
			channel={props.channel}
		/>
		<CommentsContainer className="comments-container" />
		<RelatedVideos className="related-videos" videos={props.relatedVideos} />
	</div>
);

WatchContent.propTypes = {};

const mapStateToProps = (state, props) => ({
	video: selector_videoById(state, props.match.params.videoId),
	relatedVideos: selector_relatedVideos(state, props.match.params.videoId),
	channel: selector_channelDetails(state, props.match.params.videoId),
});

const actionCreators = {
	fetchWatchDetails: watchActions.action_fetchWatchDetails.request,
};

export default withRouter(
	connect(
		mapStateToProps,
		actionCreators,
	)(WatchContent),
);
