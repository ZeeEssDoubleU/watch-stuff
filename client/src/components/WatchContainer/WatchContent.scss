.watch-grid {
   display: grid;
   grid-template: auto auto auto 1fr / auto 402px; // 1fr row takes available left over whitespace
   padding: 24px;
   grid-gap: 8px 24px;
   
   .video-container {
      grid-row: 1 / 2;
      grid-column: 1 / 2;
   }
   .metadata {
      grid-row: 2 / 3;
      grid-column: 1 / 2;
   }
   .video-info {
      grid-row: 3 / 4;
      grid-column: 1 / 2;
   }
   .comments {
      grid-row: 4 / 5;
      grid-column: 1 / 2;
   }
   .related-videos {
      grid-row: 1 / 5;
      grid-column: 2 / 3;
   }
}