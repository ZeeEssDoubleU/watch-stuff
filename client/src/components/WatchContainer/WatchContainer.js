import React, { Component } from "react";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import PropTypes from "prop-types";

import WatchContent from "./WatchContent";

import * as watchActions from "../../store/actions/watch";
import { selector_youtubeLibraryLoaded } from "../../store/reducers/api";

class WatchContainer extends Component {
	componentDidMount() {
		if (this.props.youtubeLibraryLoaded) {
			this.props.fetchWatchDetails(this.props.match.params.videoId);
		}
	}
	componentDidUpdate(prevState) {
		if (this.props.youtubeLibraryLoaded !== prevState.youtubeLibraryLoaded) {
			this.props.fetchWatchDetails(this.props.match.params.videoId);
		}
	}
	render() {
		return <WatchContent />;
	}
}

WatchContainer.propTypes = {};

const mapStateToProps = state => ({
	youtubeLibraryLoaded: selector_youtubeLibraryLoaded(state),
});

const actionCreators = {
	fetchWatchDetails: watchActions.action_fetchWatchDetails.request,
};

export default withRouter(
	connect(
		mapStateToProps,
		actionCreators,
	)(WatchContainer),
);
