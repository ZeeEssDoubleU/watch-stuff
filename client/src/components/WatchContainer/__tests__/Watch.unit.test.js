import React from "react";
import { shallow } from "enzyme";

import WatchContainer from "../WatchContainer";

test("renders WatchContainer", () => {
	const wrapper = shallow(<WatchContainer />);
	expect(wrapper).toMatchSnapshot();
});
