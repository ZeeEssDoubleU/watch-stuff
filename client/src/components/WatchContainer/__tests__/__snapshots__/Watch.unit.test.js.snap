// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders WatchContainer 1`] = `
<div
  className="watch-container"
>
  <Video
    className="video-container"
    id="vDjcWlCT8rg"
  />
  <VideoMetadata
    className="video-metadata"
    viewCount={1000}
  />
  <VideoInfo
    className="video-info-container"
  />
  <CommentsContainer
    className="comments-container"
  />
  <RelatedVideo
    className="related-videos"
  />
</div>
`;
